src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Arm.java

```diff
@@ -81,53 +90,79 @@ public Arm() {
         setArmTarget(goalState[ARM].position, 0);
         setWristTarget(goalState[WRIST].position, 0);
         wristMotor.setSmartCurrentLimit(WRIST_CURRENT_LIMIT_AMP);
+        lastArmPos = getArmPos();
+        lastMeasuredTime = Timer.getFPGATimestamp();
 
         // SmartDashboard.putNumber("Arm Max Vel", MAX_FF_VEL[ARM]);
         // SmartDashboard.putNumber("Wrist Max Vel", MAX_FF_VEL[WRIST]);
-        SmartDashboard.putNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
-        SmartDashboard.putNumber("Arm Tolerance Pos", posToleranceRad[ARM]);
-        SmartDashboard.putNumber("Wrist Tolerance Pos", posToleranceRad[WRIST]);
-        SmartDashboard.putNumber("Arm Tolerance Vel", velToleranceRadPSec[ARM]);
-        SmartDashboard.putNumber("Wrist Tolerance Vel", velToleranceRadPSec[WRIST]);
+        // SmartDashboard.putNumber("ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD", ARM_TELEOP_MAX_GOAL_DIFF_FROM_CURRENT_RAD);
+        // SmartDashboard.putNumber("Arm Tolerance Pos", posToleranceRad[ARM]);
+        // SmartDashboard.putNumber("Wrist Tolerance Pos", posToleranceRad[WRIST]);
+        // SmartDashboard.putNumber("Arm Tolerance Vel", velToleranceRadPSec[ARM]);
+        // SmartDashboard.putNumber("Wrist Tolerance Vel", velToleranceRadPSec[WRIST]);
     }
 
     @Override
     public void periodic() {
 
         if(DriverStation.isDisabled()) resetGoal();
 
+        double currTime = Timer.getFPGATimestamp();
+        SmartDashboard.putNumber("Current Time", currTime);
+        SmartDashboard.putNumber("Last Update (s)", lastMeasuredTime);
+        
+        // when the value is different
+        double currentArmPos = getArmPos();
+        if (currentArmPos != lastArmPos) {
+            lastMeasuredTime = currTime;
+            lastArmPos = currentArmPos;
+        }
+        isArmEncoderConnected = currTime - lastMeasuredTime < DISCONNECTED_ENCODER_TIMEOUT_SEC;
```

Comment:
logic for shutting off the arm when disconnected from encoder should also be copied for wrist